packages/flutter_tools/test/commands.shard/hermetic/build_ios_test.dart

```diff
@@ -281,4 +299,103 @@ void main() {
     Usage: () => usage,
     XcodeProjectInterpreter: () => FakeXcodeProjectInterpreterWithBuildSettings(),
   });
+
+  testUsingContext('Trace error if xcresult is empty.', () async {
+    final BuildCommand command = BuildCommand();
+
+    _createMinimalMockProjectFiles();
+
+    await createTestCommandRunner(command).run(
+      const <String>['build', 'ios', '--no-pub']
+    );
+
+    expect(testLogger.traceText, contains('xcresult parser: Unrecognized top level json format.'));
+  }, overrides: <Type, Generator>{
+    FileSystem: () => fileSystem,
+    ProcessManager: () => FakeProcessManager.list(<FakeCommand>[
+      xattrCommand,
+      _setUpFakeXcodeBuildHandler(onRun: () {
+        fileSystem.systemTempDirectory.childDirectory(_xcBundleFilePath).createSync();
+      }),
+      _setUpXCResultCommand(onRun: () {
+      }),
```

Comment:
`onRun` can be null.
```suggestion
      _setUpXCResultCommand(),
```